src/kv/test/kv_test.cpp

```diff
@@ -1456,13 +1454,44 @@ TEST_CASE("Conflict resolution")
   REQUIRE(res3 == kv::CommitResult::SUCCESS);
 
   REQUIRE(tx1.commit_version() > tx2.commit_version());
-  REQUIRE(tx2.get_read_version() >= tx2.get_read_version());
+  REQUIRE(tx1.get_read_version() >= tx2.get_read_version());
 
   // Re-running a _committed_ transaction is exceptionally bad
   REQUIRE_THROWS(tx1.commit());
   REQUIRE_THROWS(tx2.commit());
 }
 
+TEST_CASE("Backup conflict detection")
```

Comment:
- first tx that must not be parallel with other (next tx must wait)
- primary runs a bunch of tx with no dependency - apply order to backup have no dependencies
- - primary runs a bunch of tx with dependencies - apply in correct & incorrect order